### PR TITLE
Filter for refcheck job that excludes new files

### PR DESCRIPTION
**Pull request by NikitaSkrynnik**
Signed-off-by: Nikita Skrynnik <nikita.skrynnik@xored.com>

## Issue
https://github.com/networkservicemesh/deployments-k8s/issues/6645

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -28,13 +28,32 @@ jobs:
...
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
       - name: Convert to local and back and check refs
         run: |
           ./to-local.sh
           ./to-ref.sh
-          git diff --name-only --exit-code
+
+          added_files=$(git diff --name-only --diff-filter=A --merge-base origin/main HEAD)
```
Comment (denis-tingaikin) [inline .github/workflows/ci.yaml:47]: What if  the main branch will be changed?

I think we need compare with main revision that branch based on.

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

**Comment by denis-tingaikin** [comment]
In general looks good and correct to me, but we should use _old_ main to avoid problems with merges.

**Comment by denis-tingaikin** [comment]
Merging this because it blocks other things.

@NikitaSkrynnik Feel free to resolve comments in the separate PR

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (NikitaSkrynnik) [reply .github/workflows/ci.yaml:47]: `--merge-base origin/main HEAD` is equal to `git diff $(git merge-base origin/main HEAD) HEAD` ([docs](https://git-scm.com/docs/git-diff#:~:text=If%20%2D%2Dmerge%2Dbase%20is%20given%2C%20use%20the%20merge%20base%20of%20the%20two%20commits%20for%20the%20%22before%22%20side.%20git%20diff%20%2D%2Dmerge%2Dbase%20A%20B%20is%20equivalent%20to%20git%20diff%20%24(git%20merge%2Dbase%20A%20B)%20B.)).

I tested `git merge-base main HEAD` locally on branch from this [PR](https://github.com/networkservicemesh/deployments-k8s/pull/6647) and it gave me next result: `007815ea576e5d077bea9dc0589a1eb86ff87952`

`git log --oneline`: 
```
99dbc34c9c (HEAD -> test-linkchecker, fork/test-linkchecker) minor fixes in  refcheck
ecdf48029f add advanced refcheck
40efea1451 add more files with refs
22083ff629 add diff script
4393326afa add diff script
86cb0c4882 rename files
eebe4ed9a6 add more files
e140196d4e change ref
e43a417ced change ref
10de4deab5 add new file with ref
007815ea57 Update references
54be591950 Update go.mod and go.sum to latest version from networkservicemesh/cm…
```
From git log you can see commit `007815ea57`. So `git merge-base` shows the commit from which brunch `HEAD` grows

`git log --oneline` for main branch: 

```
24287f6d96 (HEAD -> main, main/main, main/HEAD) Update references
17bf4171f7 feat: add vl3 dns example (#6644)
54f7de4a1c Update references
78e553a66a Filter for refcheck job that excludes new files (#6677)
1df8068ee6 Update references
48a95f7731 Update references
a489573ded Update references
910069378d Update go.mod and go.sum to latest version from networkservicemesh/cm…
bc6521d484 Update references
60c873ce4f Update go.mod and go.sum to latest version from networkservicemesh/cm…
ba76ec9303 Update go.mod and go.sum to latest version from networkservicemesh/cm…
d95a7b4d9c Update go.mod and go.sum to latest version from networkservicemesh/cm…
2a82d5b160 Update references
f38e17ac86 Update go.mod and go.sum to latest version from networkservicemesh/cm…
33e7765f97 Update references
3a014dcc99 Update references
840a801783 Update go.mod and go.sum to latest version from networkservicemesh/cm…
168e6e79df Update go.mod and go.sum to latest version from networkservicemesh/cm…
50e85b172e Update references
6b5ae90e77 Update references
cce3e0f63d Update go.mod and go.sum to latest version from networkservicemesh/cm…
e711dbe462 Update go.mod and go.sum to latest version from networkservicemesh/cm…
6d7732efc0 Update references
a782ce2bfd Update go.mod and go.sum to latest version from networkservicemesh/cm…
36fed1672f Update references
f1f439c416 Update go.mod and go.sum to latest version from networkservicemesh/cm…
c527ac04dc Update go.mod and go.sum to latest version from networkservicemesh/cm…
529ee01bf4 Update references
5776d0891e Update references
5af8c3980a Update references
2e7a99fdd6 Update references
e6a8428239 Update go.mod and go.sum to latest version from networkservicemesh/cm…
47cf07bfb0 Update go.mod and go.sum to latest version from networkservicemesh/cm…
19a2e8582c Update go.mod and go.sum to latest version from networkservicemesh/cm…
d0a1d3b654 Update go.mod and go.sum to latest version from networkservicemesh/cm…
c4a00bd78d Update references
95e0019e8e Update go.mod and go.sum to latest version from networkservicemesh/cm…
830ff1c55d Update references
e0cd69f141 Add consul example UI (#6582)
d0c89a7739 Update references
43952aad0a Update references
0053b403de Update references
e3ec6c4129 Update go.mod and go.sum to latest version from networkservicemesh/cm…
c95c95cade Update go.mod and go.sum to latest version from networkservicemesh/cm…
e0ccffc189 Update go.mod and go.sum to latest version from networkservicemesh/cm…
6fe059f137 Update go.mod and go.sum to latest version from networkservicemesh/cm…
fcef486ffa Update references
fcef486ffa Update references
c1859d1fac Update references
6d2f14991d Update go.mod and go.sum to latest version from networkservicemesh/cm…
84fef98891 Update go.mod and go.sum to latest version from networkservicemesh/cm…
a77f887d6b Update references
58f0f44175 Update references
57b41547ab Update go.mod and go.sum to latest version from networkservicemesh/cm…
56bdea891e Update references
f370a280c5 Update go.mod and go.sum to latest version from networkservicemesh/cm…
e1080d7fd4 Update references
84deb100ee Update go.mod and go.sum to latest version from networkservicemesh/cm…
e03a1b7b8d Update go.mod and go.sum to latest version from networkservicemesh/cm…
bec22fa65d Update references
2c4fd33cd0 Update references
28f75ff1f7 Update references
d8d13f4f9f Update go.mod and go.sum to latest version from networkservicemesh/cm…
f9fa253760 Update go.mod and go.sum to latest version from networkservicemesh/cm…
5892c10fba Update go.mod and go.sum to latest version from networkservicemesh/cm…
b5e4e35826 Update go.mod and go.sum to latest version from networkservicemesh/cm…
661791ce64 Update go.mod and go.sum to latest version from networkservicemesh/cm…
c761e83e6f Update references
34673fac26 Update references
0861f299a5 Update go.mod and go.sum to latest version from networkservicemesh/cm…
b97f6a75b5 Update go.mod and go.sum to latest version from networkservicemesh/cm…
007815ea57 Update references
```

As you can see in updated `main` branch our commit `007815ea57` is very far from the last commit.

**Pull request merged**